### DISCUSSION TITLE
[JS Hook] Can I use `ParseAndExec` ?

### DISCUSSION THREAD

**Discussion by hungcrush** [pocketbase/pocketbase#5497] [Q&A] [answered]
Hi, I have a custom API that requires pagination. Can I use ParseAndExec on my JavaScript hook to achieve this?

Thank you so much.

**Comment by ganigeorgiev** [comment, marked as answer]
If you mean [`Provider.ParseAndExec`](https://pkg.go.dev/github.com/pocketbase/pocketbase@v0.22.20/tools/search#Provider.ParseAndExec) then No because there is no binding in the JSVM for `search.Provider` at the moment.

It is not clear how and where do you plan to use it but for retrieving paginated list of records you could check [`$app.dao().findRecordsByFilter()` helper](https://pocketbase.io/docs/js-records/#fetch-multiple-records):

```js
const records = $app.dao().findRecordsByFilter(
    "articles",                                    // collection
    "status = 'public' && category = {:category}", // filter
    "-publised",                                   // sort
    10,                                            // limit
    0,                                             // offset
    { category: "news" },                          // optional filter params
)
```

**Reply by hungcrush** [reply]
Hi~ Thank you for you reply... 

I have another question. How can I count all records by filter? I mean like `findRecordsByFilter` but return total records.

Thank you so much.

**Reply by ganigeorgiev** [reply]
There is no such method at the moment, but without details about your use case to understand the bigger picture I can't help you further.

**Reply by hungcrush** [reply]
Thank you. I realize that I can use **View Collection**, then use `onRecordsListRequest` to custom record 😬 

Thank you one more time. I love what you're doing.
